functions/src/index.ts

```diff
@@ -242,13 +328,52 @@ exports.calculateAqi = functions.pubsub
           nowCastPm25: nowcastPm25.reading,
           aqi: aqi,
         };
+        aqiBufferData = {
+          aqi: aqi,
+          timestamp: FieldValue.serverTimestamp(),
+        };
       }
-    }
+      // Send AQI reading to current-readings to be displayed on the map
+      await firestore.collection('current-reading').doc('pm25').set({
+        lastUpdated: FieldValue.serverTimestamp(),
+        data: currentData,
+      });
 
-    await firestore.collection('current-reading').doc('pm25').set({
-      lastUpdated: FieldValue.serverTimestamp(),
-      data: currentData,
-    });
+      /// / -------- Update the AQI Circular Buffer -------- ////
+
+      // Document reference for current sensor
+      const sensorDocRef = firestore.collection('/sensors').doc(docId);
+
+      await sensorDocRef.get().then(doc => {
```

Comment:
Again, you shouldn't have to have both the await keyword and then, here's a [blog post](https://www.smashingmagazine.com/2020/11/comparison-async-await-versus-then-catch/) that I found helpful

Reply:
Okay, I'll explore this and try to recreate the bug that led me to this decision after addressing the other initial comments.